### DISCUSSION TITLE
How Synchronize multi MPU6050 by I2C Multiplexer

### DISCUSSION THREAD

**Discussion by williamesp2015** [jrowberg/i2cdevlib#621] [Q&A]
I got difficulty with DMP using MPU6050_6Axis_MotionApps20.h (old and recent versions) when I want to  read 5-10 MPU605 at the same time. Is there any way to synchronize starting DMP and sampling rates?
void loop()
{
for(i=0;i<8;i++)
{
SelectI2cChannel(i);
while (!mpu.dmpGetCurrentFIFOPacket(fifoBuffer))
  {
    delayMicroseconds(100);
  }
  mpu.dmpGetQuaternion(&q, fifoBuffer);
  mpu.dmpGetGravity(&gravity, &q);
  mpu.dmpGetYawPitchRoll(ypr, &q, &gravity);
  mpu.dmpGetLinearAccel(&aaReal, &aa, &gravity);
  mpu.dmpGetGyro(&gy, fifoBuffer);
 PrintData();
}
}

**Comment by ZHomeSlice** [comment]
simple..... I hope :)

in MPU6050_6Axis_MotionApps612.cpp
you will want to break dmpInitialize() into 2 parts
the first part will start at the beginning and stop at line 367 just before the following 3 lines.
```
	I2Cdev::writeBytes(devAddr,0x6A, 1, &(val = 0xC0), wireObj); // 1100 1100 USER_CTRL: Enable Fifo and Reset Fifo
	I2Cdev::writeBytes(devAddr,0x38, 1, &(val = 0x02), wireObj); // 0000 0010 INT_ENABLE: RAW_DMP_INT_EN on
	I2Cdev::writeBit(devAddr,0x6A, 2, 1, wireObj);      // Reset FIFO one last time just for kicks. (MPUi2cWrite reads 0x6A first and only alters 1 bit and then saves the byte)

```
you will repeat this first part for each of your MPU's

The second part will execute the above 3 lines for each of your MPU's which will start the MPU DMP routine.

Z

**Comment by williamesp2015** [comment]
Many thanks. My sensors (MPU6050) return 0xC instead of 0x34 even though I modified 
uint8_t MPU6050_Base::getDeviceID() {
    I2Cdev::readBits(devAddr, MPU6050_RA_WHO_AM_I, MPU6050_WHO_AM_I_BIT, MPU6050_WHO_AM_I_LENGTH, buffer, I2Cdev::readTimeout, wireObj);
  if((buffer[0]==0x34) || (buffer[0]==0xC))
  return 0x34;
}
but it stuck at loop. So I have to use uint8_t MPU6050_6Axis_MotionApps20::dmpInitialize() in the MPU6050_6Axis_MotionApps20.cpp
Any advice would be appreciated.

**Comment by ZHomeSlice** [comment]
MPU6050_6Axis_MotionApps20.cpp is deprecated. The only advantage is a smaller binary file
use 
MPU6050_6Axis_MotionApps612.cpp instead
I have ZERO documentation on V2.0
I have tones of documentation of V6.12

Let's see if that helps fix problems
Z
